File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 9/14/2025, 9:14:56 AM
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0;
        int j = 0;
        while(j < nums.length){
            if(nums[j] != 0){
                nums[i++] = nums[j];
            }
            j++;
        }

        for(; i<nums.length; i++){
            nums[i] = 0;
        }
    }
}

File: 283-MoveZeroes/283-MoveZeroes.c
// Last updated: 9/14/2025, 9:14:59 AM
void moveZeroes(int* nums, int numsSize){
      int index=0;
      for(int i=0;i<numsSize;i++){
          if(nums[i]!=0){
              nums[index++]=nums[i];
          }
      }
      for(int i=index;i<numsSize;i++){
          nums[i]=0;
      }
}